hs_core/tasks.py

```diff
@@ -398,6 +402,26 @@ def create_bag_by_irods(resource_id):
 
     res = get_resource_by_shortkey(resource_id)
     istorage = res.get_irods_storage()
+    bag_full_name = res.get_irods_path('bags/{res_id}.zip'.format(res_id=resource_id), prepend_short_id=False)
+
+    if res.locked:
+        # resource is locked meaning its bag is being created by another celery task, wait until the lock
+        # is released which indicates the bag is available
+        while res.locked:
+            sleep(1)

```

Comment:
We shouldn't be holding up a thread and a response when a resource is locked due to bag creation.  Ideally, I think we should return the task id of the bag creation task and poll the celery task until it's done.